File: packages/ios/Sources/Components/Dialog/Dialog.figma.swift
import SwiftUI
import Figma

struct KozmosDialogBasicConnect: FigmaConnect {
    let component = KozmosDialog<EmptyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8101"
    var variant = ["Content": "Basic"]

    @FigmaString("Title Text")
    var title: String = "Edit profile"

    @FigmaString("Description Text")
    var description: String = "Make changes to your profile here."

    @FigmaString("Body Text")
    var bodyText: String = "Use dialog body content for a short task, form, or confirmation."

    var body: some View {
        KozmosDialog(
            isPresented: .constant(true),
            title: self.title,
            description: self.description,
            bodyText: self.bodyText
        )
    }
}

struct KozmosDialogFormConnect: FigmaConnect {
    let component = KozmosDialog<EmptyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8101"
    var variant = ["Content": "Form"]

    @FigmaString("Title Text")
    var title: String = "Edit profile"

    @FigmaString("Description Text")
    var description: String = "Make changes to your profile here."

    @FigmaString("Body Text")
    var bodyText: String = "Use dialog body content for a short task, form, or confirmation."

    @FigmaString("Primary Action Text")
    var primaryActionTitle: String = "Save changes"

    var body: some View {
        KozmosDialog(
            isPresented: .constant(true),
            title: self.title,
            description: self.description,
            bodyText: self.bodyText,
            primaryActionTitle: self.primaryActionTitle,
            onPrimaryAction: {}
        )
    }
}

struct KozmosDialogFooterConnect: FigmaConnect {
    let component = KozmosDialog<EmptyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8101"
    var variant = ["Content": "Footer"]

    @FigmaString("Title Text")
    var title: String = "Review changes"

    @FigmaString("Description Text")
    var description: String = "Confirm the details before continuing."

    @FigmaString("Body Text")
    var bodyText: String = "Use dialog body content for a short task, form, or confirmation."

    @FigmaString("Primary Action Text")
    var primaryActionTitle: String = "Save changes"

    @FigmaString("Secondary Action Text")
    var secondaryActionTitle: String = "Cancel"

    var body: some View {
        KozmosDialog(
            isPresented: .constant(true),
            title: self.title,
            description: self.description,
            bodyText: self.bodyText,
            primaryActionTitle: self.primaryActionTitle,
            secondaryActionTitle: self.secondaryActionTitle,
            onPrimaryAction: {},
            onSecondaryAction: {}
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/Dialog/Dialog.figma.kt
package com.kozmos.components.dialog

import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.figma.code.connect.FigmaVariant

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=143-13060")
@FigmaVariant("Content", "Basic")
class KozmosDialogBasicConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Edit profile"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Make changes to your profile here."

    @FigmaProperty(FigmaType.Text, "Body Text")
    val bodyText: String = "Use dialog body content for a short task, form, or confirmation."

    @Composable
    fun ComponentExample() {
        KozmosDialog(
            onDismissRequest = {},
            title = title,
            description = description,
            bodyText = bodyText
        )
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=143-13060")
@FigmaVariant("Content", "Form")
class KozmosDialogFormConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Edit profile"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Make changes to your profile here."

    @FigmaProperty(FigmaType.Text, "Body Text")
    val bodyText: String = "Use dialog body content for a short task, form, or confirmation."

    @Composable
    fun ComponentExample() {
        KozmosDialog(
            onDismissRequest = {},
            title = title,
            description = description,
            bodyText = bodyText,
            primaryActionText = "Save changes",
            onPrimaryAction = {}
        )
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=143-13060")
@FigmaVariant("Content", "Footer")
class KozmosDialogFooterConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Review changes"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Confirm the details before continuing."

    @FigmaProperty(FigmaType.Text, "Body Text")
    val bodyText: String = "Use dialog body content for a short task, form, or confirmation."

    @Composable
    fun ComponentExample() {
        KozmosDialog(
            onDismissRequest = {},
            title = title,
            description = description,
            bodyText = bodyText,
            primaryActionText = "Save changes",
            secondaryActionText = "Cancel",
            onPrimaryAction = {},
            onSecondaryAction = {}
        )
    }
}
